frontend/website/src/components/chatroom.svelte

```diff
@@ -1,40 +1,76 @@
 <script lang="ts">
+	import { pipe, subscribe, type Subscription } from "wonka";
+	import { client } from "$lib/gql";
+	import { graphql } from "$/gql";
 	import CloseSidebar from "./closeSidebar.svelte";
-	import { user } from "../store/user";
+	import { user } from "$/store/user";
 	import { onMount } from "svelte";
+	import { MessageType, type ChatMessage, type User } from "$/gql/graphql";
+	import {} from "os";
 
 	export let collapsed = false;
+	export let channelId: string;
+	let svelteId = 1;
 
 	function collapseNav() {
 		collapsed = !collapsed;
+		if (collapsed) {
+			unsubscribeFromMessages();
+		} else {
+			subscribeToMessages();
+		}
+	}
+
+	function createSystemMessage(content: string): ChatMessage {
+		return {
+			authorId: "",
+			// We cast to User because this is a system message and we don't care about the author or the channel
+			channel: {} as User,
+			author: {} as User,
+			channelId: channelId,
+			createdAt: "",
+			id: "",
+			type: MessageType.System,
+			content: content,
```

Comment:
leaving it as empty strings and stuff feels odd, could it be `null` instead? feel like that would be nicer